### PR TITLE
Added libvmaf + support for it in ffmpeg-full

### PR DESCRIPTION
**Pull request by cfsmp3**
#70593 #### Things done

- Created package libvmaf (Netflix's video quality library)
- Added myself as a mantainer for it
- Added libvmaf support to ffmpeg-full

Tested on Ubuntu Linux. Nothing fancy here though, all quite standard.

###### Notify maintainers

cc @jonringer @Fuuzetsu

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg build libvmaf ffmpeg-full

**Comment by cfsmp3** [comment]
@jonringer  Fuuzetsu Anything else needed from me? (or is there another reviewer I should tag)

**Comment by jonringer** [comment]
LGTM, I've just been busy. Sorry @marsam does it look good to you?

File: pkgs/development/libraries/libvmaf/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "v${version}";
+    sha256="10kgcdf06hzhbl5r7zsllq88bxbyn282hfqx5i3hkp66fpq896d2";
+  };
+
+  nativeBuildInputs = [ autoconf automake intltool libtool pkgconfig ];
+  outputs = [ "out" "dev" ];
+  doCheck = true;
+
+  makeFlags = [ "INSTALL_PREFIX=${placeholder "out"}" ];
+
```
Comment (marsam) [inline pkgs/development/libraries/libvmaf/default.nix]: I think you need to add:
```nix
  postFixup = ''
    substituteInPlace "$dev/lib/pkgconfig/libvmaf.pc" \
      --replace "includedir=/usr/local/include" "includedir=$dev"
  '';
```
for pkgconfig to find libvmaf

File: pkgs/development/libraries/libvmaf/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/Netflix/vmaf";
+    description = "Perceptual video quality assessment based on multi-method fusion (VMAF).";
```
Comment (marsam) [inline pkgs/development/libraries/libvmaf/default.nix]: description [should not include a period at the end](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes):
```suggestion
    description = "Perceptual video quality assessment based on multi-method fusion (VMAF)";
```

**Review by marsam — CHANGES_REQUESTED** [review]
please disable libvmaf for ffmpeg-full on darwin

```diff
diff --git i/pkgs/top-level/all-packages.nix w/pkgs/top-level/all-packages.nix
index 70bd668e4af..7cb69f15a40 100644
--- i/pkgs/top-level/all-packages.nix
+++ w/pkgs/top-level/all-packages.nix
@@ -10802,6 +10802,7 @@ in
     game-music-emu = if stdenv.isDarwin then null else game-music-emu;
     libjack2 = if stdenv.isDarwin then null else libjack2;
     libmodplug = if stdenv.isDarwin then null else libmodplug;
+    libvmaf = if stdenv.isDarwin then null else libvmaf;
     openal = if stdenv.isDarwin then null else openal;
     libmfx = if stdenv.isDarwin then null else intel-media-sdk;
     libpulseaudio = if stdenv.isDarwin then null else libpulseaudio;
```
Besides that, LTGM

**Comment by cfsmp3** [comment]
@marsam sorry for the delay in fixing these things, I've been away a few days. All done.

**Comment by marsam** [comment]
@GrahamcOfBorg build libvmaf ffmpeg-full

**Comment by cfsmp3** [comment]
@marsam if it looks good, could you merge? thanks

**Comment by marsam** [comment]
Sorry for the delay. I've disabled libvmaf for aarch64, I hope you don't mind

@GrahamcOfBorg build libvmaf ffmpeg-full

**Comment by marsam** [comment]
@GrahamcOfBorg buil ffmpeg-full

**Comment by marsam** [comment]
@GrahamcOfBorg build ffmpeg-full

**Pull request merged**